packages/components/src/form-toggle/test/index.tsx

```diff
@@ -0,0 +1,109 @@
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle, { noop } from '..';
+import type { FormToggleProps } from '../types';
+
+const getInput = () => screen.getByRole( 'checkbox' ) as HTMLInputElement;
+
+const ControlledFormToggle = ( { onChange }: FormToggleProps ) => {
+	const [ isChecked, setChecked ] = useState( false );
+	return (
+		<FormToggle
+			checked={ isChecked }
+			onChange={ ( value ) => {
+				setChecked( ( state ) => ! state );
+				onChange( value );
+			} }
+		/>
+	);
+};
+
+describe( 'FormToggle', () => {
+	describe( 'basic rendering', () => {
+		it( 'should render', () => {
+			render( <FormToggle onChange={ noop } /> );
+			expect( getInput() ).not.toBeNull();
+		} );
+
+		it( 'should render a span element with an unchecked checkbox', () => {
+			const { container } = render( <FormToggle onChange={ noop } /> );
+
+			expect( getInput() ).toHaveProperty( 'checked', false );
+			expect( container.firstChild ).toMatchSnapshot();
+		} );
+
+		it( 'should render a checked checkbox and change the accessibility text to On when providing checked prop', () => {
+			render( <FormToggle onChange={ noop } checked /> );
+
+			expect( getInput() ).toHaveProperty( 'checked', true );
+		} );
+
+		it( 'should render with an additional className', () => {
+			const { container: containerDefault } = render(
+				<FormToggle onChange={ noop } />
+			);
+
+			const { container: containerWithClassName } = render(
+				<FormToggle onChange={ noop } className="testing" />
+			);
+
+			// Expect the diff snapshot to be mostly about the className.
+			expect( containerDefault ).toMatchDiffSnapshot(
+				containerWithClassName
+			);
+		} );
+
+		it( 'should render an id prop for the input checkbox', () => {
+			const { container: containerDefault } = render(
+				<FormToggle onChange={ noop } />
+			);
+
+			const { container: containerWithID } = render(
+				// Disabled because of our rule restricting literal IDs, preferring
+				// `withInstanceId`. In this case, it's fine to use literal IDs.
+				// eslint-disable-next-line no-restricted-syntax
+				<FormToggle onChange={ noop } id="test" />
+			);
+
+			// Expect the diff snapshot to be mostly about the ID.
+			expect( containerDefault ).toMatchDiffSnapshot( containerWithID );
+		} );
+	} );
+
+	describe( 'Value', () => {
+		it( 'should flip the checked property when clicked', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+
+			let state = false;
+			const setState = jest.fn( () => ( state = ! state ) );
+
+			render( <ControlledFormToggle onChange={ setState } /> );
+
+			const input = getInput();
+
+			await user.click( input );
+			expect( input ).toHaveProperty( 'checked', true );
+			expect( state ).toBe( true );
```

Comment:
I feel like this test deviates a little bit from the _exact_ contract we want to guarantee, which is:

1. The `onChange` is called when the toggle is clicked.
2. The first arg passed to the `onChange` call should be the Event.

The second point I think is the most important to test for because it is the current behavior, and if anyone changes that behavior it should sound alarms as a breaking change. And this behavior is notably different from other (newer?) components, which simply pass the new value as the first arg instead of the Event. (And LOL that the readme was wrong about this too! 😂)

So, while I do think a lot of consumers are writing their `onChange` functions like your example, I think we can provide better coverage by assuming that some consumers are using the Event in their `onChange` functions.

How about something like this?

```jsx
			const onChange = jest.fn();
			render( <ControlledFormToggle onChange={ onChange } /> );

			const input = getInput();

			await userEvent.click( input );
			expect( onChange.mock.calls[ 0 ][ 0 ].target ).toBeInTheDocument(); // no strong opinion on what this assertion should be, but something that reasonably asserts that it's an Event

			// and so on...
```

Reply:
Seem like an good idea to me. Have updated the test.